nodeJS/express/views.md

```diff
@@ -78,10 +78,41 @@ Start the server and go to the `/` route in the browser. You should see:
 EJS rocks!
 ```
 
-If you inspect the HTML in the browser's dev tools, you can see HTML is structures exactly like how we wrote the EJS template with the `message` variable replaced with its value.
+If you inspect the HTML in the browser's dev tools, you can see the HTML is structured exactly like how we wrote the EJS template with the `message` variable replaced with its value.
 
 When you hit the `/` route, `res.render("index", { message: "EJS rocks!" });` is the line that sends back the response. Since we've already defined the `views` and `view engine` app properties, the first argument of `res.render` is programmed to look for "a template called index in the specified folder", while the second argument is an object of variables that are to be made available to that specific template.
 
+### The locals variable in EJS
+
+In the example above, how did the template file know about the `message` variable? The answer is that we passed a `locals` object as the second argument to `res.render()`, which contained a `message` property. This makes it available for template engines to access local variables. We could have also set local variables onto the [`res.locals`](https://expressjs.com/en/5x/api.html#res.locals) object in our route, like so:
+
+```javascript
+app.get("/", (req, res) => {
+  res.locals.message = "EJS rocks!";
+  res.render("index");
+});
+```
+
+Then, EJS stores this in an object which is named `locals` by default (check the localsName bullet point in the Options section in the [EJS docs](https://ejs.co/#docs) to learn more). Thus the `message` variable is accessible with either `locals.message` or simply `message`.
```

Comment:
Just a wording suggestion that I think *might* make the point clearer that EJS `locals` just has access to anything in Express's `res.locals` *and* anything we pass in via `res.render`. The current wording feels like you have to pick one or the other. I don't think we'd really need a code block for the `res.locals` thing with this, nor link to the EJS docs about the `localsName` thing, since I *highly* doubt many learners will be trying to configure the EJS engine itself.

The wording below *is* just a suggestion, so feel free to play around with it if you wish, otherwise let me know what you think.

```suggestion
In the example above, how did the template file know about the `message` variable? When we render the view, EJS has access to any properties from the object we pass into `res.render`, as well as any properties on [Express's res.locals object](https://expressjs.com/en/5x/api.html#res.locals) (`res.locals` can be useful if you need to pass values to the view in one middleware function, but won't call `res.render` until later in the middleware chain).

EJS will store these properties in an object called `locals`, which you can access in the view. Similarly to the global `window` object in browsers, this allows you to access the `message` variable in the view via `locals.message`, or simply just `message`.
```

Reply:
Your suggestion sounds good to me, I suppose it's better to focus on EJS rather than the specifics of setting local variables in Express. And it's probably best to not confuse the learner by adding too much information. The analogy with the global `window` object is also a good call.